internal/service/resourcepolicy/resource.go

```diff
@@ -0,0 +1,175 @@
+package resourcepolicy
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"regexp"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/config"
+	"go.mongodb.org/atlas-sdk/v20240805003/admin"
+)
+
+const resourceName = "resource_policy"
+
+var _ resource.ResourceWithConfigure = &resourcePolicyRS{}
+var _ resource.ResourceWithImportState = &resourcePolicyRS{}
+
+var (
+	errorCreate = "error creating resource"
```

Comment:
can be const instead of var

recommend to show the resource name, e.g.: https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/internal/service/mongodbemployeeaccessgrant/resource.go#L18